system/src/system_threading.cpp

```diff
@@ -94,7 +94,7 @@ namespace std {
         thread_startup startup;
         startup.call = base.get();
         startup.started = false;
-        if (os_thread_create(&_M_id._M_thread, "std::thread", OS_THREAD_PRIORITY_DEFAULT, invoke_thread, &startup, 1024*3)) {
+        if (os_thread_create(&_M_id._M_thread, "std::thread", OS_THREAD_PRIORITY_DEFAULT, invoke_thread, &startup, 1024*4)) {
```

Comment:
This will increase the stack size on the Photon too. I think that until we know the size increase is needed there, that this should be conditional. You can check `HAL_PLATFORM_CLOUD_UDP` which will be 1 when UDP is enabled. 

I'd like to know what is requiring so much stack size in the UDP handshake, and I'm curious why we get corruption since the heap grows from the bottom of memory while the stack grows from the top. Ah...but the system thread stack is allocated on the heap so that's why any overflow corrupts the heap.